src/query/sql/src/planner/binder/table.rs

```diff
@@ -260,10 +260,13 @@ impl Binder {
                     }
                     parent = bind_context.parent.as_mut();
                 }
-                return Err(ErrorCode::UnknownTable(format!(
-                    "Unknown table `{database}`.`{table_name}` in catalog '{catalog}'"
-                ))
-                .set_span(*span));
+                if e.code() == 1025 {
+                    return Err(ErrorCode::UnknownTable(format!(
+                        "Unknown table `{database}`.`{table_name}` in catalog '{catalog}'"
+                    ))
+                    .set_span(*span));
+                }
+                return Err(e);
```

Comment:
Need help cc @lichuang 

<img width="1227" alt="image" src="https://github.com/datafuselabs/databend/assets/41979257/9f9bbec7-8e26-4d89-af47-8090d0f4db0f">

The error is from `resolve_data_source()`.